src/components/Store.ts

```diff
@@ -24,44 +35,115 @@ export class Store {
         return this.store
     }
 
-    getInstances() : { [key : string] : IInstance } {
-        return this.context.globalState.get<{
-            [key : string] : IInstance;
-        }>(InfluxDBInstancesKey) || {}
+    // Get the config. Don't cache this, otherwise we'd have to set up change
+    // event listeners.
+    get config() : vscode.WorkspaceConfiguration {
+        return vscode.workspace.getConfiguration('vsflux')
+    }
+
+    async getInstances() : Promise<{ [key : string] : IInstance }> {
+        const dataSource = this.config.get<string>('datasource', 'db')
+        if (dataSource === 'cli') {
+            const { stdout, stderr } = await exec('influx config list --json')
+            if (stderr) {
+                console.error(`Error from influx cli: ${stdout}`)
+                vscode.window.showErrorMessage('Error invoking the influx command')
+                return {}
+            }
+
+            const cliData : { [key : string] : CLIInstance } = JSON.parse(stdout)
+            const instances : { [key : string] : IInstance } = {}
+            for (const [id, config] of Object.entries(cliData)) {
+                instances[id] = {
+                    id,
+                    disableTLS: false,
```

Comment:
I didn't see any tls disabling option in the CLI config so changing this option in vsflux wouldn't do anything anymore. Is that ok? Should remove it or at least disable it for CLI configs if so.

Reply:
Yeah, I'm not entirely sure how to handle that just yet. Honestly, I'm not all that stoked about the `disableTLS` functionality in the first place, and if I were to see the bug as filed today, I'd probably just say "no."

Reply:
Tweaking the template for the edit view to omit the `disableTLS` checkbox when the cli config is used might be the way to go then?

Reply:
Tweaked this template to not even should the checkbox when disabling TLS isn't available.